rplugin/python3/deoplete/deoplete.py

```diff
@@ -68,96 +64,109 @@ def gather_candidates(self, context):
-            self.profile_end(source.name)
-            if context['candidates'] and isinstance(
-                    context['candidates'][0], str):
-                # Convert to dict
-                context['candidates'] = [{'word': x}
-                                         for x in context['candidates']]
-
-            ignorecase = context['ignorecase']
-            smartcase = context['smartcase']
-            camelcase = context['camelcase']
+        for source_name, source in self.itersource(context):
             try:
+                if source.disabled_syntaxes and 'syntax_names' not in context:
+                    context['syntax_names'] = get_syn_names(self.__vim)
+                ctx = copy.deepcopy(context)
+                charpos = source.get_complete_position(ctx)
+                if charpos >= 0 and source.is_bytepos:
+                    charpos = bytepos2charpos(
+                        ctx['encoding'], ctx['input'], charpos)
+                ctx['complete_str'] = ctx['input'][charpos:]
+                ctx['complete_position'] = charpos2bytepos(
+                    ctx['encoding'], ctx['input'], charpos)
+                ctx['max_abbr_width'] = min(source.max_abbr_width,
+                                            ctx['max_abbr_width'])
+                ctx['max_menu_width'] = min(source.max_menu_width,
+                                            ctx['max_menu_width'])
+                if ctx['max_abbr_width'] > 0:
+                    ctx['max_abbr_width'] = max(20, ctx['max_abbr_width'])
+                if ctx['max_menu_width'] > 0:
+                    ctx['max_menu_width'] = max(10, ctx['max_menu_width'])
+
+                if charpos < 0 or self.is_skip(ctx, source.disabled_syntaxes,
+                                               source.min_pattern_length,
+                                               source.max_pattern_length,
+                                               source.input_pattern):
+                    # Skip
+                    continue
+
+                # Gathering
+                self.profile_start(ctx, source.name)
+                ctx['candidates'] = source.gather_candidates(ctx)
+                self.profile_end(source.name)
+
+                if 'candidates' not in ctx or not ctx['candidates']:
+                    continue
+
+                if ctx['candidates'] and isinstance(ctx['candidates'][0], str):
+                    # Convert to dict
+                    ctx['candidates'] = [{'word': x}
+                                         for x in ctx['candidates']]
+
+                # Filtering
+                ignorecase = ctx['ignorecase']
+                smartcase = ctx['smartcase']
+                camelcase = ctx['camelcase']
+
                 # Set ignorecase
-                if (smartcase or camelcase) and re.search(
-                        r'[A-Z]', context['complete_str']):
-                    context['ignorecase'] = 0
+                if (smartcase or camelcase) \
```

Comment:
Okay I'll change it.

The reason I do it is because I don't like the alternative:

```python
if (smartcase or camelcase) and re.search(r'[A-Z]',
                                          ctx['complete_str']):
    ctx['ignorecase'] = 0
```

Without braces, I think it makes it harder to read because the lines appear to be separated by whitespace.  The whitespace separation increases if the far right function has a lot of arguments.